### DISCUSSION TITLE
Add new @CreateRequestContextAsync() decorator

### DISCUSSION THREAD

**Discussion by mehedimi** [mikro-orm/mikro-orm#4805] [General]
**Is your feature request related to a problem? Please describe.**
When I use `@CreateRequestContext()` decorator, I need the MikroORM instance but If I have an ORM instance wrapped with Promise then It won't work.

Ex:
```ts
class DB {
  static conn: MikroORM;

  static async get(): Promise<MikroORM> {
    if (DB.conn) {
      return DB.conn;
    }

    return DB.create();
  }

  static async create(): Promise<MikroORM> {
    // Creating a new connection, set the connection instance to `conn` property and returns it
  }
}

class NewJob {

  // This won't work
  @CreateRequestContext(() => DB.get())
  async handle() {
    const conn = await DB.get()
    //
  }
}
```

**Describe the solution you'd like**
If MikroORM has `@CreateRequestContextAsync()` decorator which resolves the Promise first and then registers the new request context, It would be very helpful. 

**Describe alternatives you've considered**
Create a new `@CreateRequestContextAsync()` decorator for handling the Promise<MikroORM> instance.

**Comment by B4nan** [comment]
So you want the callback for `@CreateRequestContext ` decorator to be awaited? Should be just about adding the `await` somewhere here:

https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/decorators/CreateRequestContext.ts#L9

PR welcome for that.

Also FYI you can use `MikroORM.initSync` in v6, the connections are created lazily when needed in there.

**Reply by mehedimi** [reply]
Great! I am working on it!
